common/archiver/filestore/visibilityArchiver.go

```diff
@@ -340,5 +340,6 @@ func convertToExecutionInfo(record *visibilityRecord) *shared.WorkflowExecutionI
 		SearchAttributes: &shared.SearchAttributes{
 			IndexedFields: archiver.ConvertSearchAttrToBytes(record.SearchAttributes),
 		},
+		IsWorkflowOpen: common.BoolPtr(record.CloseStatus.Ptr() == nil),
```

Comment:
I don't think you need this check. Visibility records are archived after retention period so the workflow must be closed.

Reply:
removed.